batch/test/jwt-test-user.json

```diff
@@ -0,0 +1,5 @@
+{"id": -1,
```

Comment:
I'm confused here.  There's no id field in the user-jwt secrets on the cluster.  Looks like there is on id field of any kind.  Don't we need a way to identify users?  @akotlar what was the intention here?

Reply:
I think the point was to verify that a given jwt could be decoded and that protected routes were only accessible when the token was present (the main test that uses the token tests the authentication decorator). 

@danking it would be useful for the complementary test to be run (no token, check ValueError)

Reply:
I'm talking about the structure of the token, not it's existence.  It doesn't look like real ones, and looking at the real ones, I'm realizing I don't know what to compare to see if two users are the same.

Reply:
This is a fake json I cooked up for testing purposes, the schema of the token has changed a few times since I made this and I have repeatedly not updated this. Probably should generate this the same way as the real tokens. For now, I'll remove the id field.

Reply:
Agreed.  I can cook up steps in ci2 to create a Hail account signing key, a test users database, and Hail accounts for testing.

But we still want some form of id, right?  For comparing users.

Reply:
will all users have a gsa_email ?

Reply:
I believe so, because I think we plan on using the service account strategy for handling user authentication to google services (even if one day we move to using the user's own google accounts, this strategy is still valid for server-server applications)

Reply:
This doesn't seem like a good idea.  We won't have a gsa if we are running on AWS.  There is an auto increment integer id column in the user table.  We should include that in the jwt.

Unfortunately, MySQL (InnoDB, specifically) can reuse values.  I seems much better if they are globally unique, although fixing this isn't urgent.